### PR TITLE
Timepicker issue

### LINKED ISSUE #1267: Start Time and End Time time pickers are not working in Firefox
Currently, in the createEvent.html file, the user is not able to click on the time pickers to input a start time and an end time. In other words, upon clicking the icons, nothing happens. 

image

### PR DESCRIPTION
**Pull request by ali-ramazani**
Fixes issue #1267

We fixed the issue with the start time and end time buttons not triggering the timepicker dropdown attached to the input field. The issue was the timepicker method in the timepicker object didn't expect "show" as argument. Instead, it expected "open". Moreover, we prevent the event handler from propagating up because it would close the dropdown as in the implementation of the timepicker closes the dropdown if it's not trigged from the input field directly.

### CODE REVIEW

**Comment by thealphagurlux** [comment]
To add on to Ali's comment, we also implemented an if else statement into the function bound to the button because the button would open the time picker scroll but clicking it again would not close the scroll box, so we used the time pickers 'closed' method in order to check if the scroll box was open or not, and close and open it appropriately.

File: app/static/js/createEvents.js [hunk 1]
```diff
@@ -173,7 +162,16 @@ $(document).ready(function () {
     if ($(this).val() && $("#endDatePicker-" + $(this).data("page-location")).val()) {
       calculateRecurringEventFrequency();
     }
-  });
+  })
+  // Time picker click events
+  $(".timeIcons").on("click", function(e) {
+    e.stopPropagation()
+    console.log($(this).siblings(".timepicker").timepicker("closed") );
```
Comment (BrianRamsay) [inline app/static/js/createEvents.js]: An extra console.log

File: app/static/js/createEvents.js [hunk 2]
```diff
@@ -173,7 +162,16 @@ $(document).ready(function () {
...
+    if ($(this).siblings(".timepicker").timepicker("closed"))
+      $(this).siblings(".timepicker").timepicker("open"); 
+    else 
+    $(this).siblings(".timepicker").timepicker("close")
```
Comment (BrianRamsay) [inline app/static/js/createEvents.js]: This should be indented. Also let's use braces around the blocks, even though they are single line

File: app/templates/admin/createEvent.html [hunk 3]
```diff
@@ -26,62 +26,65 @@
...
+  <!-- Include jQuery -->
+  <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+  <!-- Include jQuery UI (required for datepicker) -->
+  <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
+  <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+  <!-- Include jQuery Timepicker -->
+  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/timepicker/1.3.5/jquery.timepicker.min.css">
+  <script src="//cdnjs.cloudflare.com/ajax/libs/timepicker/1.3.5/jquery.timepicker.min.js"></script>
+  <!-- Custom Styles -->
+  <link href="https://cdn.jsdelivr.net/gh/kartik-v/bootstrap-fileinput@5.5.0/css/fileinput.min.css" media="all" rel="stylesheet" type="text/css" />
```
Comment (BrianRamsay) [inline app/templates/admin/createEvent.html]: This stylesheet doesn't need to be in the scripts block.

File: app/templates/admin/createEvent.html [hunk 4]
```diff
@@ -26,62 +26,65 @@
...
 {% set tabName = 'edit' %}
 
-
 {% block scripts %}
-  {{super()}}
+  {{super()}} 
+  <!-- Include jQuery -->
+  <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+  <!-- Include jQuery UI (required for datepicker) -->
+  <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
```
Comment (BrianRamsay) [inline app/templates/admin/createEvent.html]: This shouldn't work here, but maybe the browser is letting us get away with it. It should go in the styles block if necessary.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

File: app/templates/admin/createEvent.html (hunk 3 above)
Comment (JohnCox2211) [reply app/templates/admin/createEvent.html]: moved into style block

File: app/templates/admin/createEvent.html (hunk 4 above)
Comment (JohnCox2211) [reply app/templates/admin/createEvent.html]: moved into style block

File: app/static/js/createEvents.js (hunk 1 above)
Comment (JohnCox2211) [reply app/static/js/createEvents.js]: removed

File: app/static/js/createEvents.js (hunk 2 above)
Comment (JohnCox2211) [reply app/static/js/createEvents.js]: fixed formatting

**Comment by JohnCox2211** [comment]
Corrected things mentioned in comments

File: app/templates/admin/createEvent.html [hunk 5]
```diff
@@ -452,6 +451,7 @@ <h5 class="modal-title" id="renewModel">Renew Event</h5>
     </div>
   </div>
 </form>
+{% endblock %}
```
Comment (bledsoef) [inline app/templates/admin/createEvent.html:454]: This endblock is messing up styling on the page.

File: app/static/js/createEvents.js [hunk 6]
```diff
@@ -136,7 +134,23 @@ function storeMultipleOfferingEventAttributes() {
     let entries = [];
     $(".extraSlots").children().each(function(index, element) {
         let rowData = $.map($(element).find("input"), (el) => $(el).val());
+  // Initialize timepicker
```
Comment (bledsoef) [inline app/static/js/createEvents.js:137]: Indentation on this timepicker block of code is weird and the .extraSlots function does not close, causing some errors on my end.

File: app/templates/admin/createEvent.html (hunk 5 above)
Comment (bledsoef) [reply app/templates/admin/createEvent.html:454]: It was initially just styling but now it is breaking completely for me

File: app/static/js/createEvents.js [hunk 7]
```diff
@@ -136,7 +134,23 @@ function storeMultipleOfferingEventAttributes() {
...
+  $('.timepicker').timepicker({
+    timeFormat: 'h:mm p',
+    interval: 15,
+    minTime: '12:00am',
+    maxTime: '11:45pm',
+    defaultTime: '11:00am',
```
Comment (bledsoef) [inline app/static/js/createEvents.js:143]: I am now hitting errors, but before I was I saw that the time for both time pickers was 11:00 am. We want to by default have the times an hour apart. For instance start time is noon and end time is 1pm.

**Review by bledsoef — CHANGES_REQUESTED** [review]
Looks like merge conflicts might have caused some issues.

**Comment by bledsoef** [comment]
There are some awful merge conflicts. Going to close this and reimplement the changes in development.

**Pull request closed without merging**